app/models/service/resource_linking.rb

```diff
@@ -0,0 +1,36 @@
+module Service::ResourceLinking
+  extend ActiveSupport::Concern
+
+  def add_provider_vms(options)
+    # required keys in options hash:
+    #   :target_id, the id of the provider to refresh
+    #   :array_of_uid_ems, an array of VMs' uid_ems
```

Comment:
Does the external caller to the API need to know this name?  Suggest using `uid_ems_array`.

Reply:
Whoever calls this method needs to pass in the two parameters: ```target_id``` and ```uid_ems_array```.